### ISSUE TITLE
[local development bug] yarn build generates /lib/index.js instead of /lib/src/index.js

### ISSUE THREAD

**Issue by molecula451** [ubiquity/ubiquibot#796]
just did a clean install and I get another error upon yarn start: Cannot find module './lib/src/index.js'. it seems Typescript is generating ./lib/index.js instead of ./lib/src/index.js. Not sure why it changed

Originally from telegram: by @whilefoo https://t.me/UbiquityDAO/31132/31145

**Comment by pavlovcik** [comment]
It is unclear to me if this is a blocker, or if this issue is easy to circumvent

**Comment by molecula451** [comment]
> It is unclear to me if this is a blocker, or if this issue is easy to circumvent

yeah it is a blocker

**Comment by molecula451** [comment]
@pavlovcik on a clean build, touching nothing
```typescript
$ probot run ./lib/src/index.js
/home/ubiquibot/node_modules/resolve/lib/sync.js:111
    var err = new Error("Cannot find module '" + x + "' from '" + parent + "'");
              ^
Error: Cannot find module './lib/src/index.js' from '/home/ubiquibot'
    at resolveSync (/home/ubiquibot/node_modules/resolve/lib/sync.js:111:15)
    at resolveAppFunction (/home/ubiquibot/node_modules/probot/lib/helpers/resolve-app-function.js:34:23)
    at combinedApps (/home/ubiquibot/node_modules/probot/lib/run.js:101:79)
    at async Server.load (/home/ubiquibot/node_modules/probot/lib/server/server.js:60:9)
    at async run (/home/ubiquibot/node_modules/probot/lib/run.js:105:9) {
  code: 'MODULE_NOT_FOUND'
}
```

PRing...

**Comment by devpanther** [comment]
> It is unclear to me if this is a blocker, or if this issue is easy to circumvent

It is not, this was an issue a month ago, you can see the discussion here https://github.com/ubiquity/ubiquibot/pull/558#issuecomment-1663936287

Cause of the problem: https://github.com/ubiquity/ubiquibot/pull/558#issuecomment-1661891217

So it was changed from `./lib/index.js` to `./lib/src/index.js` to solve the issue but I guess the json has been removed so its gone back to normal

**Comment by molecula451** [comment]
you may see the clean output from the npm above

**Comment by molecula451** [comment]
`index.js` magically moved out `/src`

**Comment by molecula451** [comment]
thanks for the input @devpanther

**Comment by devpanther** [comment]
> `index.js` magically moved out `/src`

it was magic lol, you can read through the comment I attached and the StackOverflow on one of the comment

**Comment by molecula451** [comment]
i know, just for the fun :]

**Comment by molecula451** [comment]
impressive the stuff at [#583 Comment](https://github.com/ubiquity/ubiquibot/pull/558#issuecomment-1663936287) that it's how bug emerges

**Comment by molecula451** [comment]
https://github.com/ubiquity/ubiquibot/pull/797#issuecomment-1731109294

**Comment by molecula451** [comment]
```typescript
yarn run v1.22.19
$ probot run ./lib/index.js

  _|    _|  _|_|_|    _|_|_|    _|_|      _|    _|  _|_|_|  _|_|_|_|_|  _|      _|
  _|    _|  _|    _|    _|    _|    _|    _|    _|    _|        _|        _|  _|
  _|    _|  _|_|_|      _|    _|  _|_|    _|    _|    _|        _|          _|
  _|    _|  _|    _|    _|    _|    _|    _|    _|    _|        _|          _|
    _|_|    _|_|_|    _|_|_|    _|_|  _|    _|_|    _|_|_|      _|          _|

    
INFO (server): Running Probot v12.3.1 (Node.js: v18.16.1)
INFO (server): Forwarding https://smee.io/q5fmDornulyf5ge to http://localhost:3000/
INFO (server): Listening on http://localhost:3000
INFO (server): Connected
```

**Comment by Keyrxng** [comment]
I'm not getting these issues with a fresh yarn install

**Comment by seprintour** [comment]
You can't, you have to delete the lib folder first

**Comment by molecula451** [comment]
> ```ts
> probot run ./lib/index.js
> ```

this is reproducible, you can see how probot looks for `index.js` in `/lib` not in `/lib/src`

`probot run ./lib/index.js`

**Comment by molecula451** [comment]
> > ```ts
> > probot run ./lib/index.js
> > ```
> 
> this is reproducible, you can see how probot looks for `index.js` in `/lib` not in `/lib/src`
> 
> `probot run ./lib/index.js`

refer to https://github.com/ubiquity/ubiquibot/pull/558#issuecomment-1663936287 for the bug context

**Comment by molecula451** [comment]
all feedback it's great, altho everything points out` index.js` should be and is at `/lib/index.js` an start there

**Comment by whilefoo** [comment]
@pavlovcik it seems the permit was generated and even saved to the DB, from that moment on it should post all comments but it only posted conversation reward comment. My suspicion is on Netlify's 10 second timeout.

**Comment by molecula451** [comment]
> @pavlovcik it seems the permit was generated and even saved to the DB, from that moment on it should post all comments but it only posted conversation reward comment. My suspicion is on Netlify's 10 second timeout.

10 sec?

**Comment by pavlovcik** [comment]
@0xcodercrane Also suspects a timeout issue. I think it makes sense because of the logs showing that it will generate a payment permit as well as the inconsistency of it sometimes posting. 

Logical errors would be consistent

**Comment by devpanther** [comment]
> @0xcodercrane Also suspects a timeout issue. I think it makes sense because of the logs showing that it will generate a payment permit as well as the inconsistency of it sometimes posting. 
> 
> Logical errors would be consistent 

You think #794 would fix this?, it joins all the permits together in 1

**Comment by molecula451** [comment]
> @pavlovcik it seems the permit was generated and even saved to the DB, from that moment on it should post all comments but it only posted conversation reward comment. My suspicion is on Netlify's 10 second timeout.

both? assignee and creator permits?

**Comment by 0x4007** [comment]
> > @0xcodercrane Also suspects a timeout issue. I think it makes sense because of the logs showing that it will generate a payment permit as well as the inconsistency of it sometimes posting.
> > Logical errors would be consistent
> 
> You think #794 would fix this?, it joins all the permits together in 1

Depends on why its timing out. The only way I could see this being the solution is if each comment waits for the previous network response in series. Then it could be, for example, 100ms for each network request, but in series thats like 400ms of lag. 

@0xcodercrane claims that its due to the `mdast` comment parser which doesn't make sense to me.

**Comment by molecula451** [comment]
but what the logs says? it doesn't make sense time out either, there is comments rewards there to be honest

**Comment by 0x4007** [comment]
😉

**Comment by wannacfuture** [comment]
The fix isn't merged to production, yet.

**Comment by molecula451** [comment]
you should know?

**Comment by molecula451** [comment]
> @molecula451 please be sure to review this conversation and implement any necessary fixes. Unless this is closed as completed, its payment of **21.15 WXDAI** will be deducted from your next bounty.

the only permit for this amount was  devpanther: [ CLAIM 21.15 WXDAI ]

**Comment by 0x4007** [comment]
https://gnosisscan.io/tx/0xa7b16eba70bf94ecfad02df3b2fcc1f571fa58a7a152ec4c2c98592b072a7be6
